syntax/ast.mjs

```diff
@@ -105,18 +105,21 @@ export class NumberLiteral extends Expression {
 }
 
 export class MessageReference extends Expression {
-    constructor(id) {
+    constructor(id, attribute) {
         super();
         this.type = "MessageReference";
         this.id = id;
+        this.attribute = attribute;
     }
 }
 
 export class TermReference extends Expression {
-    constructor(id) {
+    constructor(id, attribute, args) {
         super();
         this.type = "TermReference";
         this.id = id;
+        this.attribute = attribute;
+        this.args = args;
```

Comment:
Naming things... `attribute` or `attr` to make it consistent with `id` and `args`?

Reply:
Full name sounds about right to me. `attr` is less obvious about the fact that this is exactly one or none.